test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -439,20 +436,28 @@ class test_uct_cm_sockaddr : public uct_test {
             self->server_accept(self->m_server, conn_request,
                                 server_connect_cb, server_disconnect_cb, self);
         }
+        UCS_TEST_MESSAGE << "Connection requested";
+        self->m_cm_state |= TEST_CM_STATE_CONNECT_REQUESTED;
```

Comment:
why do you need this flag set here?

Reply:
to avoid race conditions on `m_delayed_conn_reqs`, for example.

Reply:
https://github.com/openucx/ucx/pull/4520 should fix the race.

Reply:
are you sure it fixes wait_for_bits(TEST_CM_STATE_CONNECT_REQUESTED) case?
IMO, `self->m_cm_state |= TEST_CM_STATE_CONNECT_REQUESTED;` and `self->m_server_recv_req_cnt++;` should go together after adding request to the queue

Reply:
yes, it should fix. please take a look at the change in that pr.

Reply:
I don't see how it can fix `m_cm_state` check

Reply:
why not?

Reply:
because flag is set before request is queued

Reply:
but there is a fence there between them

Reply:
Yes, this is exactly why it doesn't work - flag is set before request is queued.